detox/android/detox/src/full/java/com/wix/detox/reactnative/idlingresources/asynctask/DetoxBusyResource.kt

```diff
@@ -0,0 +1,45 @@
+package com.wix.detox.reactnative.idlingresources.asynctask
+
+import androidx.test.espresso.IdlingResource
+import com.wix.detox.reactnative.idlingresources.IdlingResourceDescription
+
+class DetoxBusyResource {
+    var resourceMap = HashMap<String, Any>()
+
+    constructor(resource: IdlingResource? = null) {
+        resource?.let{
+            val detoxBusyResourceHelper = DetoxBusyResourceHelper()
+            resourceMap.putAll(detoxBusyResourceHelper.formatResource(it))
+        }
+    }
+
+    constructor(idlingResourceDescription: IdlingResourceDescription?) {
+        idlingResourceDescription?.json()?.let {
+            resourceMap.putAll(it)
+        }
+    }
```

Comment:
Feels weird to have this just for testing. Would it make more sense to use `constructor(IdlingResource)` with a mocked idling resource?